### PR TITLE
Fix : IEEE 754 compliance for SIMD math ops and correct scalar fallback

### PR DESCRIPTION
**Pull request by CokieMiner**
### Bugs fixed

These functions previously produced incorrect results for non-finite inputs (NaN, ±∞, ±0). All fixes verified against IEEE 754-2008 §9.2 and §6.2.

| Function | Old behavior | Fixed behavior |
|----------|-------------|----------------|
| `sin_cos` / `sin` / `cos` / `tan` | NaN/±∞ → garbage (polynomial chain on invalid range reduction) | NaN/±∞ → `(NaN, NaN)` |
| `exp` | `exp(NaN)` → `0.0`, `exp(+∞)` → `0.0` | `exp(NaN)` → NaN, `exp(+∞)` → +∞ |
| `exp` | `exp(overflow)` (e.g. 800.0) → `0.0` | → +∞ |
| `exp2` | Same bugs as `exp` | Same fixes |
| `ln` | `ln(±0)` → `+∞` (wrong sign) | → `-∞` |
| `ln` | `ln(-∞)` → `-∞` | → NaN (invalid operation) |

### Prerequisite fix: `is_zero_or_subnormal` bitmask

`is_zero_or_subnormal` used `i64x4::round_float(simd_eq(...))` which converts the `-1` comparison mask arithmetically to `-1.0_f64` (`0xBFF0000000000000`). This accidentally worked with `blend` (which only checks the sign bit), but the new IEEE 754 fixes use bitwise `&`/`|` patterns that require true bitmask values. Fixed by replacing `round_float` with `cast::<_, f64x4>(mask)`, a bitwise reinterpret that produces `0xFFFFFFFFFFFFFFFF` for true lanes. Applied to all 6 float types.

### `ln` underflow threshold

`VM_SMALLEST_NORMAL` in `f64*_*.rs` was `1.17549435E-38`, the f32 smallest normal, instead of `2.2250738585072014E-308` (the f64 value). This caused `ln` to incorrectly flag all values below 1.17e-38 as underflow.

### Known limitation

`is_zero_or_subnormal()` does not distinguish `exponent==0 && mantissa==0` from `exponent==0 && mantissa!=0`, so subnormals (~1e-308..2.2e-308) are treated as zero in `ln`. Both old and new code share this behavior, old returned `+∞`, new returns `-∞`. True subnormal inputs are essentially never encountered in practice. Documented with a comment.

### Tests added

Non-finite edge case tests for `sin_cos`, `exp`, `exp2`, `ln` in `t_f64x4.rs`, covering NaN, ±∞, ±0, and finite overflow.

### Files changed

```
src/f64x2_.rs
src/f64x4_.rs
src/f64x8_.rs
src/f32x4_.rs
src/f32x8_.rs
src/f32x16_.rs
tests/all_tests/t_f64x4.rs
```

All 955 existing tests pass.

### CODE REVIEW

**Comment by CokieMiner** [comment]
Wtf I didn't request this shit

**Comment by CokieMiner** [comment]
ok they were valided so fixed

**Comment by Lokathor** [comment]
I've never seen CoPilot review a PR like this before. It seems to be a new setting that's on by default. I've turned it off. I'll have a look at this in a little bit.

**Comment by CokieMiner** [comment]
Actually my bad here, for some reason the automatic reviews were turned on on my account, I don't remeber turning them on but whatever.

File: tests/all_tests/t_f64x4.rs [hunk 1]
```diff
@@ -827,6 +855,15 @@ fn impl_f64x4_ln() {
       assert!(diff_from_std[0] < 0.00000000001);
     }
   }
+
+  // Edge cases
+  let edge = f64x4::from([f64::NAN, f64::INFINITY, f64::NEG_INFINITY, 0.0]);
+  let res = edge.ln();
+  println!("res.to_array() = {:?}", res.to_array());
```
[the remark this answers is not in the document]
Comment (Lokathor) [reply tests/all_tests/t_f64x4.rs]: Copilot, this suggestion is explicitly incorrect.

**Comment by Lokathor** [comment]
I think everyone got it turned on across all of GitHub sometime recently. Microsoft is pushy like that.
